concepts/05 Widgets/DataGrid/25 Sorting/20 API/1 Initial and Runtime Sorting.md

```diff
@@ -18,60 +19,215 @@ Rows are sorted according to the data source by default. Set the [sortOrder](/ap
+                sort-order="asc"
+            />
+            <DxColumn
+                :sort-index="0"
+                data-field="Country"
+                sort-order="asc"
+            />
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxDataGrid, DxColumn } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DataGrid, Column } from 'devextreme-react/data-grid';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <DataGrid>
+                    <Column
+                        sortIndex={1}
+                        dataField="City"
+                        sortOrder="asc" />
+                    <Column
+                        sortIndex={0}
+                        dataField="Country"
+                        sortOrder="asc" />
+                </DataGrid>
+            );
+        }
+    }
+    export default App;
+
 ---
 
 Change the **sortOrder** and **sortIndex** options using the [columnOption](/api-reference/10%20UI%20Widgets/GridBase/3%20Methods/columnOption(id_optionName_optionValue).md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Methods/#columnOptionid_optionName_optionValue') method to sort at runtime.
 
 ---
 ##### jQuery
 
-    <!--JavaScript-->var dataGrid = $("#dataGridContainer").dxDataGrid("instance");
+    <!--JavaScript-->
+    var dataGrid = $("#dataGridContainer").dxDataGrid("instance");
     dataGrid.columnOption("Country", {
         sortOrder: "desc",
         sortIndex: 2
     });
 
 ##### Angular
+    <!--HTML-->
+    <dx-data-grid ... >
+        <dxi-column
+            dataField="Country"
+            [(sortIndex)]="sortIndex"
+            [(sortOrder)]="sortOrder">
+        </dxi-column>
+    </dx-data-grid>
 
     <!--TypeScript-->
-    import { ..., ViewChild } from "@angular/core";
-    import { DxDataGridModule, DxDataGridComponent } from "devextreme-angular";
-    // ...
+    import { DxDataGridModule } from "devextreme-angular";
+    
     export class AppComponent {
-        @ViewChild(DxDataGridComponent, { static: false }) dataGrid: DxDataGridComponent;
-        // Prior to Angular 8
-        // @ViewChild(DxDataGridComponent) dataGrid: DxDataGridComponent;
-        sortByCountries () {
-            this.dataGrid.instance.columnOption("Country", {
-                sortOrder: "desc",
-                sortIndex: 2
-            });
+        sortIndex: number = 0;
+        sortOrder: string = "asc";
+
+        sortByCountries(index: number, order: string) {
+            this.sortIndex = index;
+            this.sortOrder = order;
```

Comment:
The names of the binding properties should somehow indicate that they are used for the Country column only. I suggest `countrySortIndex` and `countrySortOrder`.

Besides, it is peculiar that the method demands an `index`, which shouldn't be necessary to apply sorting. I would expect that the required index would be somehow calculated by the method.

Reply:
We still need to handle onOptionChanged in React, as other default sorting changes need to be processed as well

Reply:
I agree with countrySortIndex and countrySortOrder.  

As for the method, in jQuery we set options via columnOption:
`dataGrid.columnOption("Country", {
        sortOrder: "desc",
        sortIndex: 2
    });`
Above, we wrote: _Change the **sortOrder** and **sortIndex** options_ 
As you can see, we simply set these options at runtime without additional calculations.

In Angular/React/Vue, we decided not to use a widget instance and columnOption, so I added a simple method that performs the same operation when it is invoked.

If the method signature may mislead the customer, we can simplify it:
`sortByCountries()
            this.countrySortIndex = 2;
            this.countrySortOrder = 'desc';
        }`

Reply:
Let's change the sortOrder only - the last unocuppied sortIndex will be set automatically.